CMakePresets.json

```diff
@@ -88,140 +121,73 @@
       }
     },
     {
-      "name": "x64-debug",
-      "displayName": "x64 Debug",
-      "description": "Windows x64 Debug",
-      "inherits": "x64",
+      "name": "debug-build",
+      "displayName": "Debug Build",
+      "description": "Debug Build (Hidden)",
       "hidden": true,
       "cacheVariables": {
         "CMAKE_BUILD_TYPE": "Debug"
       }
     },
     {
-      "name": "x64-release",
-      "displayName": "x64 Release",
-      "description": "Windows x64 Release",
-      "inherits": "x64",
+      "name": "release-build",
+      "displayName": "Release Build",
+      "description": "Release Build (Hidden)",
       "hidden": true,
       "cacheVariables": {
         "CMAKE_BUILD_TYPE": "RelWithDebInfo"
       }
     },
     {
-      "name": "x86-debug",
-      "displayName": "x86 Debug",
-      "description": "Windows x86 Debug",
-      "inherits": "x86",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "Debug"
-      }
-    },
-    {
-      "name": "x86-release",
-      "displayName": "x86 Release",
-      "description": "Windows x86 Release",
-      "inherits": "x86",
+      "name": "enable-tests",
+      "displayName": "Enable Tests",
+      "description": "Enable Tests (hidden)",
       "hidden": true,
       "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
-      }
-    },
-    {
-      "name": "x64-static-debug",
-      "displayName": "Windows x64 Debug",
-      "description": "Windows x64 Debug",
-      "inherits": "x64-static",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "Debug"
-      }
-    },
-    {
-      "name": "x64-static-release",
-      "displayName": "Windows x64 Release, static",
-      "description": "Windows x64 Release, static",
-      "inherits": "x64-static",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "RelWithDebInfo"
+        "BUILD_TESTING": true,
+        "ENABLE_PROXY_TESTS": false,
+        "BUILD_SAMPLES": true
```

Comment:
Do we always want to couple building samples when enabling tests?

For example, we previously weren't building samples for the `x86-static-debug-tests` preset, but now we will, because of
> "inherits": [ "x86-static", "debug-build", "enable-tests" ]

Reply:
We could absolutely add samples as a separate dimension. 

Different question: When would we want to build tests without samples?

Reply:
Usually I build tests so I can run them, to validate correctness in my inner dev loop/locally. That's pretty common.

Since we tend not to have breaking changes, samples are typically fairly stable, and I am happy with CI running them to make sure they are written correctly and don't accidentally break. The times where I care to build samples is much lower than tests.

Reply:
That's reasonable. I'll make that change.